crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -500,3 +732,125 @@ pub(crate) async fn deserialize_move_struct(
 
     Ok((struct_tag, move_struct))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::str::FromStr;
+
+    #[test]
+    fn test_owner_filter_intersection() {
+        let f0 = ObjectFilter {
+            owner: Some(SuiAddress::from_str("0x1").unwrap()),
+            ..Default::default()
+        };
+
+        let f1 = ObjectFilter {
+            owner: Some(SuiAddress::from_str("0x2").unwrap()),
+            ..Default::default()
+        };
+
+        assert_eq!(f0.clone().intersect(f0.clone()), Some(f0.clone()));
+        assert_eq!(f0.clone().intersect(f1.clone()), None);
+    }
+
+    #[test]
+    fn test_key_filter_intersection() {
+        let i1 = SuiAddress::from_str("0x1").unwrap();
+        let i2 = SuiAddress::from_str("0x2").unwrap();
+        let i3 = SuiAddress::from_str("0x3").unwrap();
+        let i4 = SuiAddress::from_str("0x4").unwrap();
+
+        let f0 = ObjectFilter {
+            object_ids: Some(vec![i1, i3]),
+            object_keys: Some(vec![
+                ObjectKey {
+                    object_id: i2,
+                    version: 1,
+                },
+                ObjectKey {
+                    object_id: i4,
+                    version: 2,
+                },
+            ]),
+            ..Default::default()
+        };
+
+        let f1 = ObjectFilter {
+            object_ids: Some(vec![i1, i2]),
+            object_keys: Some(vec![ObjectKey {
+                object_id: i4,
+                version: 2,
+            }]),
+            ..Default::default()
+        };
+
+        let f2 = ObjectFilter {
+            object_ids: Some(vec![i1, i3]),
+            ..Default::default()
+        };
+
+        let f3 = ObjectFilter {
+            object_keys: Some(vec![
+                ObjectKey {
+                    object_id: i2,
+                    version: 2,
+                },
+                ObjectKey {
+                    object_id: i4,
+                    version: 2,
+                },
+            ]),
+            ..Default::default()
+        };
+
+        assert_eq!(
+            f0.clone().intersect(f1.clone()),
```

Comment:
For these test cases, it should yield the result if we flip the intersection, ie f1.intersect.f0 right

Reply:
Correct